packages/module-source/src/module-source.js

```diff
@@ -100,3 +113,5 @@ export function ModuleSource(source, opts = {}) {
   this.__needsImportMeta__ = needsImportMeta;
   freeze(this);
 }
+
+Object.setPrototypeOf(ModuleSource.prototype, AbstractModuleSource.prototype);
```

Comment:
Doesn't this approach also require setting the prototype of `ModuleSource` itself to `AbstractModuleSource`?
```js
class B { static name = "B"; }
class A extends B { static name = "A"; }
assert.equal(Object.getPrototypeOf(A.prototype), B.prototype);
assert.equal(Object.getPrototypeOf(A), B);
```

or are we anticipating an unusual relationship between the two?

Reply:
Negative. Pushed changes including tests for the property and prototype subgraph shape and hardening.